python/lsst/afw/cameraGeom/camera.py

```diff
@@ -28,24 +28,35 @@
 class Camera(DetectorCollection):
     """!A collection of Detectors that also supports coordinate transformation
     """
-    def __init__(self, name, detectorList, transformMap):
+    def __init__(self, name, detectorList, transformMap, pupilFactoryClass):
         """!Construct a Camera
 
         @param[in] name  name of camera
         @param[in] detectorList  a sequence of detectors in index order
         @param[in] transformMap  a CameraTransformMap whose native system is FOCAL_PLANE
             and that at least supports PUPIL coordinates
+        @param[in] pupilFactoryClass  a PupilFactory class for this camera
         """
         self._name = name
         self._transformMap = transformMap
         self._nativeCameraSys = self._transformMap.getNativeCoordSys()
+        self._pupilFactoryClass = pupilFactoryClass
         super(Camera, self).__init__(detectorList)
 
     def getName(self):
         """!Return the camera name
         """
         return self._name
 
+    def getPupilFactory(self, *args, **kwargs):
```

Comment:
I assume you are trying to insulate against changes in the underlying API in the future, but since you call out the positional arguments in the docstring, I think it would be good to put them in the signature.  You can still say `def getPupilFactory(self, visitInfo, pupilSize, npix, *args, **kwargs)` if you want to allow additional positional and keyword arguments.

Another option may be to use the `@wraps` decorator from `functools` to forward the docstring from the factory class.